pkg/awsSGSetup/aws_security_group_setup.go

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) 2020 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awssgsetup
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"regexp"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("AWS_SG_Setup")
+var TRACE = 7
+var DEBUG = 5
+
+// setupAWSSecurityGroups updates the master and worker security groups used in an
+// OpenShift AWS setup. It sets a time that should be checked before attempting
+// to do another call to setup.
+func SetupAWSSecurityGroups(ctx context.Context, client client.Client) error {
```

Comment:
no tests? :scream_cat:

Reply:
Yeah this is pretty dependent on the APIs k8s client and more so the AWS go sdk. It would either require mocking those which would probably be a lot of work or isolating the API clients from the other code and only testing the "other" code which would be fairly minimal.